### DISCUSSION TITLE
请问有适配Mac端的计划吗？

### DISCUSSION THREAD

**Discussion by rockliuxn** [SydneyOwl/senhaix-freq-writer-enhanced#12] [General]
首先，感谢@SydneyOwl给HAM分享了这么好的一个工具。 ☕️

如题，Mac用户表示非常期望能使用这款工具！ 🙏

**Comment by SydneyOwl** [comment]
感谢支持！
近日正在尝试适配linux端和mac端，但由于原软件使用winform开发，同时所用的.net framework版本过低不支持跨平台，可能还需要一段时间确认是否能够成功适配。

**Comment by rockliuxn** [comment]
感谢辛苦付出，为爱发电！☕️
期待多平台版本早日上线！🎉🎉

**Comment by SydneyOwl** [comment]
@rockliuxn 
您好！目前森海克斯8800/8600的macOS写频软件正在测试中，如果您的设备是8600/8800且macOS版本大于10.15的话能否帮忙测试下？由于软件没有进行公证（签名），打开时需要覆盖安全设置是正常现象
release地址：https://github.com/SydneyOwl/senhaix-freq-writer-enhanced/releases/tag/v0.2.2-alpha

**Comment by rockliuxn** [comment]
@SydneyOwl  大神太给力了！ 👍👍

我的设备：
SHX8800:  硬件版本：0.0.5，软件版本 1.8.6
Macbook M2 Pro： macOS 14.0

我回去测试一下，然后把结果发上来。

**Comment by rockliuxn** [comment]
@SydneyOwl  

刚才在Mac上安装了这个包： SHX8X00_osx_arm64.zip

在执行了 ``` xattr -cr /Applications/SHX8X00.app ``` 之后，还是打不开，报这个错误：

![image](https://github.com/SydneyOwl/senhaix-freq-writer-enhanced/assets/7591665/1d6328d7-f8ba-46ad-974a-b13e0fc2066a)

**Comment by SydneyOwl** [comment]
@rockliuxn 抱歉，请下载64位那个，刚刚在m2的机器上试了下，arm的编译出了点问题，但是64位那个是可以的
![image](https://github.com/SydneyOwl/senhaix-freq-writer-enhanced/assets/84191144/70370674-7827-4434-ad77-be3d24d53299)

**Comment by rockliuxn** [comment]
@SydneyOwl  另外一个可以打开，和您发的图片一样，等下回去拿对讲机试一下，另外这个版本支持蓝牙链接吗？

**Comment by SydneyOwl** [comment]
这个版本还不支持蓝牙，c#暂时没有找到合适的跨平台蓝牙库，后续可能会加入

**Comment by SydneyOwl** [comment]
@rockliuxn 新版本实现了windows的蓝牙写频，但经过测试mac和linux端的蓝牙功能均不完整或有bug，需要等待上游依赖修复了。
另外想请问下：mac端是否能够正常使用写频线读写频？以及是否需要额外安装驱动？

**Comment by rockliuxn** [comment]
@SydneyOwl  我身边刚好没有扩展坞，无法插对讲机的USB K头数据线，网购了一款，等明天到了我再测试一下。

**Comment by rockliuxn** [comment]
@SydneyOwl    Mac端的写频软件我这边用写频线粗略的测试了一下

### 总结
- 读频、写频都可以，也不需要安装额外的驱动
- 读频和写频这两个动作不能在软件打开后的同一个运行时操作(读频后再写频)，必须关闭后重新打开才可以，一个运行时只能进行一个操作；
- 写频是打开了读频后保存的频道列表dat文件后再执行写频操作， 这样才能成功，个人猜测：每次读频or写频后端口被占用了
- 蓝牙功能暂时无法使用

### USB端口（两个皆可使用）
 ![shx8800-usb](https://github.com/SydneyOwl/senhaix-freq-writer-enhanced/assets/7591665/0e612646-9e52-471b-b24c-25d67caf8cac)

### USB 读频成功
![shx8800-usb-success](https://github.com/SydneyOwl/senhaix-freq-writer-enhanced/assets/7591665/9a917e02-71e6-4012-9387-c7c63eabb564)

### USB 写频报错
![shx8800-usb-bug](https://github.com/SydneyOwl/senhaix-freq-writer-enhanced/assets/7591665/ba6c5419-0494-42bc-b7cb-6b4985d33882)

### 蓝牙接口报错
![shx8800-bt-bug](https://github.com/SydneyOwl/senhaix-freq-writer-enhanced/assets/7591665/f1c18ecd-16de-4068-88c5-180069ac1b48)

**Comment by SydneyOwl** [comment]
@rockliuxn 非常感谢您的反馈！我后续再完善下，另外您写频时可以用sudo试试，或者下载下最新版本，在linux上好像没有出现端口占用的问题。
蓝牙功能目前只能在windows上使用，还在等待上游依赖修复。

**Comment by SydneyOwl** [comment]
该issue将被转移到discussion中，感谢您的帮助！
